### ISSUE TITLE
Invalid authentication

### ISSUE THREAD

**Issue by DanneG** [veulsan/addon-energiinfo#4]
I am a customer of BTEA, using the URL: https://btea.energiinfo.se/ or https://btea.energiinfo.se/v3/,
but I encounter the same problem with both.

"Invalid authentication"

Logging in via their website works.
Any tips?

**Comment by veulsan** [comment]
Hej Danne,Jag har själv BTEA och har reverse engineerst deras API. Ett tips är att se till att siteid är korrekt, detta ser man enkelt genom att i utvecklarläget i firefox titta på vad som postas till inloggningssidan. Är tyvärr på resande fot och hemma först på Måndag och kan hjälpa till. Du kan ju testa slå på debug läget på modulen i homeassistant också för att få mer info.Med vänlig hälsning, Ulrik Sannsell22 maj 2024 kl. 14:22 skrev DanneG ***@***.***>:﻿
I am a customer of BTEA, using the URL: https://btea.energiinfo.se/ or https://btea.energiinfo.se/v3/,
but I encounter the same problem with both.
"Invalid authentication"
Logging in via their website works.
Any tips?

**Comment by veulsan** [comment]
Jag använder URL https://api4.energiinfo.se <https://api4.energiinfo.se/> den adressen ser man man om man slår på utvecklar läget i sin webbläsare. Då kan man se URL och tittar man på inloggningen ser man då även site ID som för mig är 13.


￼

Med vänlig hälsning,

Ulrik

**Comment by DanneG** [comment]
Körde på samma URL och siteid och det funkade fint för mig med

**Comment by DanneG** [comment]
Jag har fjärrvärme men den funkar inte, finns det möjlighet att det går att läsa ut den med?

**Comment by veulsan** [comment]
Ja, det skulle jag nog tro har dock inte byggt in ngt stöd för det i min Home assistant modul. Är ju enkelt och testa själv med python


pip install eller pip3 install

Sedan nedan kod i test.py, ändra användarnamn/lösenord och kör med python3 test.py. Är nog så att utility och signals är annorlunda för dig när man använder fjärrvärme eftersom jag reverse engineerat det själv utifrån hemsidan så vet jag inte vad som kommer in den vägen. Du kommer se skillnaden när metering_points printas ut så kan du sedan anpassa det för det jag utkommenterat under.


[{'meteringpoint_id': ’1111111', 'alias': ’ABC123', 'customer_id': ’11111111', 'customer_key': ’111111', 'utility': 'Electricity', 'signals': ['ActiveEnergy'], 'site_id': '13', 'type': 'single', 'values_type': 'consumption', 'extra': {'ean_key': ’xxxxxxxxxxxxxx', 'local_key': '50514500', 'netarea_code': 'BRG', 'address': '', 'product_key': ’XXXXXXXX'}}]

period_data = api.get_period_values(’ 1111111', '20240601', 'ActiveEnergy', ’day')

from energiinfo.api import EnergiinfoClient
from datetime import datetime

siteid='13'
apiurl='https://api4.energiinfo.se'
username='username'
password='password'
# Initialize the api, this will not make an external call
api = EnergiinfoClient(apiurl,siteid)

# Authenticate against external api using username and password,
token = api.authenticate(username,password)
if api.getStatus() == "ERR":
  errorMessage = api.getErrorMessage();
  print(f"ErrorMessage: {errorMessage}")
  exit(0)
else:
    print(f"Successfully logged in token={token}")

meter_list = api.get_metering_points()
print(meter_list)
print("====== period 20240601 ======")
#period_data = api.get_period_values(’ 1111111', '20240601', 'ActiveEnergy', ’day')
#print(period_data)
